File: llm_generated_code/ChatGPT/Python/Level1/Q1.py
def max_earnings(n, m, brands, prices):
    from collections import defaultdict
    import heapq

    # Create a dictionary to store the prices of each brand
    brand_prices = defaultdict(list)
    for i in range(m):
        brand_prices[brands[i]].append(prices[i])
    
    # Create a list to store the total earnings for each brand
    max_brand_earnings = []
    for bp in brand_prices.values():
        heapq.heapify(bp)
        max_brand_earnings.append(sum(bp))
    
    # Sort the brand earnings in descending order
    max_brand_earnings.sort(reverse=True)
    
    # Select the top n brands to maximize earnings
    max_earnings = sum(max_brand_earnings[:n])
    
    return max_earnings

def main():
    t = int(input())
    results = []
    for _ in range(t):
        n, m = map(int, input().split())
        brands = []
        prices = []
        for _ in range(m):
            bi, ci = map(int, input().split())
            brands.append(bi)
            prices.append(ci)
        
        result = max_earnings(n, m, brands, prices)
        results.append(result)
    
    for res in results:
        print(res)

if __name__ == "__main__":
    main()

File: llm_generated_code/ChatGPT/CPP/Level1/Q1.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
#include <algorithm>

using namespace std;

long long maxEarnings(int n, int k, vector<int>& brands, vector<int>& costs) {
    unordered_map<int, vector<int>> brandCostMap;
    
    // Group the costs by their brand
    for (int i = 0; i < k; ++i) {
        brandCostMap[brands[i]].push_back(costs[i]);
    }
    
    vector<long long> maxEarnings;
    
    // Find the total cost for each brand and sort
    for (auto& entry : brandCostMap) {
        vector<int>& costList = entry.second;
        long long brandTotal = 0;
        for (int cost : costList) {
            brandTotal += cost;
        }
        maxEarnings.push_back(brandTotal);
    }
    
    // Sort earnings in descending order
    sort(maxEarnings.rbegin(), maxEarnings.rend());
    
    // Take the top 'n' earnings
    long long totalEarnings = 0;
    for (int i = 0; i < min(n, (int)maxEarnings.size()); ++i) {
        totalEarnings += maxEarnings[i];
    }
    
    return totalEarnings;
}

int main() {
    int t;
    cin >> t;
    while (t--) {
        int n, k;
        cin >> n >> k;
        vector<int> brands(k), costs(k);
        for (int i = 0; i < k; ++i) {
            cin >> brands[i];
        }
        for (int i = 0; i < k; ++i) {
            cin >> costs[i];
        }
        
        cout << maxEarnings(n, k, brands, costs) << endl;
    }
    return 0;
}
